### PR TITLE
#79

### PR DESCRIPTION
**Pull request by evscott**
This pull request contains a nasty history, as I merged in the changes that have been made to master over the course of today to keep my branch current. The primary changes I've made are to the Redux store, and to the server in controller/tripRoutes and handlers/tripHandler

- Trip Actions created
- Trip Reducers created
- Root reducer created to combine multiple reducers
- Get trip info added to server as a new POST request

To look at the changes I've made look at:

- server/controller/tripRoutes
- server/handlers/tripHandler
- src/redux/actions/*
- src/redux/reducers/*

### CODE REVIEW

File: src/redux/reducers/rootReducer.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { combineReducers } from 'redux';
+import { authReducer } from './authReducers';
+import { tripReducer } from './tripReducers';
+import * as TripActions from '../actions/tripActions';
```
Comment (morgankbender) [inline src/redux/reducers/rootReducer.js]: I really like this notation. I've only seen it as
`import TripActions from '../actions/tripActions'`
which seems to function the same way, but this is way more readable

File: src/redux/reducers/tripReducers.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  UPDATE_TRIP_REQUEST,
+  UPDATE_TRIP_FAILURE,
+  UPDATE_TRIP_SUCCESS,
+  DELETE_TRIP_REQUEST,
+  DELETE_TRIP_FAILURE,
+  DELETE_TRIP_SUCCESS,
+  GET_TRIP_INFO_REQUEST,
+  GET_TRIP_INFO_FAILURE,
+  GET_TRIP_INFO_SUCCESS
+} from '../actions/tripActions';
```
Comment (morgankbender) [inline src/redux/reducers/tripReducers.js]: I feel like this should be done the same way as in rootReducer. ie 
`import * as TripActions from '../actions/tripActions';`

File: src/redux/reducers/rootReducer.js (hunk 1 above)
Comment (evscott) [reply src/redux/reducers/rootReducer.js]: This is a cool feature that's new to Javascript es6. I'm not sure if all browsers support it, yet... But I like it wayyy more too.

File: src/redux/reducers/tripReducers.js (hunk 2 above)
Comment (evscott) [reply src/redux/reducers/tripReducers.js]: I agree. Will change this now.

**Pull request closed without merging**